### PR TITLE
feat: initial release process

### PR DESCRIPTION
**Pull request by CloudBeard**
# Created Initial Release Process
This pr adds the following functionality to setup and aid the release process.
- Creates and saves logs action
- Goreleaser action to creating binary releases
- Goreleaser config for configuring goreleaser action
-  Release process documentation

The config file helps setup goreleaser. It will determine the build binaries and the changelog messages. Currently set to append so it will add each piece to the same changelog for a release. (Similar to Zarf and others)

To do a release you need to have [permissions ](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules#about-tag-protection-rules) to push tags which will trigger the GitHub goreleaser action.

```console
git tag -s vX.X.X -m "Release Commit Message"
git push origin vX.X.X
```

Closes Issue https://github.com/defenseunicorns/go-oscal/issues/46

### CODE REVIEW

**Comment by brandtkeller** [comment]
I didn't see anything out of order currently. Updated repository permissions to add the `lula-dev` team as maintainers and therefor capable of releasing protected tags.

Would like @mike-winberry approval before merge.

**Review by mike-winberry — APPROVED** [review]
Looks good!

**Review by brandtkeller — APPROVED** [review]

**Pull request merged**
